### PR TITLE
Fix issue with License Serialization - Issue 408 and 409

### PR DESCRIPTION
**Pull request by mr-zepol**
This fixes:
https://github.com/CycloneDX/cyclonedx-core-java/issues/409
https://github.com/CycloneDX/cyclonedx-core-java/issues/408

There is still some code simplification that can be done, but given the urgency and the need for a solution, it'll be done in a different PR

### CODE REVIEW

File: src/main/java/org/cyclonedx/util/serializer/LicenseChoiceSerializer.java [hunk 1]
```diff
@@ -28,48 +28,108 @@
...
+
+        if (CollectionUtils.isNotEmpty(l.getProperties())) {
+          toXmlGenerator.writeFieldName("properties");
+          toXmlGenerator.writeStartObject();
+
+          for (Property property : l.getProperties()) {
+            toXmlGenerator.writeObjectField("property", property);
+          }
+          toXmlGenerator.writeEndObject();
+        }
```
Comment (nscuro) [inline src/main/java/org/cyclonedx/util/serializer/LicenseChoiceSerializer.java:102]: I think we need to account for older spec versions here.

`acknowledgement` is new as of v1.6, `licensing` and `properties` is new as of v1.5, and so on.

For example, changing `BomXmlGeneratorTest#testIssue408Regression` to generate and validate for v1.5 fails with:

```
org.cyclonedx.exception.ParseException: cvc-complex-type.3.2.2: Attribute 'acknowledgement' is not allowed to appear in element 'license'.
org.cyclonedx.exception.ParseException: cvc-complex-type.3.2.2: Attribute 'acknowledgement' is not allowed to appear in element 'expression'.
```

Comment (nscuro) [reply src/main/java/org/cyclonedx/util/serializer/LicenseChoiceSerializer.java:102]: Thanks @mr-zepol! 

But shouldn't parsing a v1.6 BOM, and generating a v1.5 BOM based on that not yield a v1.5-compatible BOM? The tests currently assert that parsing a v1.5 yields a valid v1.5 BOM, and parsing a v1.6 yields a valid v1.6 BOM.

Parsing `issue408.xml` (v1.6) and generating a v1.5 BOM from that gives a `license` node with `acknowledgment` attribute, which fails v1.5 validation.

Comment (mr-zepol) [reply src/main/java/org/cyclonedx/util/serializer/LicenseChoiceSerializer.java:102]: > Thanks @mr-zepol!
> 
> But shouldn't parsing a v1.6 BOM, and generating a v1.5 BOM based on that not yield a v1.5-compatible BOM? The tests currently assert that parsing a v1.5 yields a valid v1.5 BOM, and parsing a v1.6 yields a valid v1.6 BOM.
> 
> Parsing `issue408.xml` (v1.6) and generating a v1.5 BOM from that gives a `license` node with `acknowledgment` attribute, which fails v1.5 validation.

I get it, you are right, thinking about it I think the filter for each version is not applied for the serializer, I will check how can I do this at least for the license and later for the other ones, thanks

Comment (mr-zepol) [reply src/main/java/org/cyclonedx/util/serializer/LicenseChoiceSerializer.java]: > Thanks @mr-zepol!
> 
> But shouldn't parsing a v1.6 BOM, and generating a v1.5 BOM based on that not yield a v1.5-compatible BOM? The tests currently assert that parsing a v1.5 yields a valid v1.5 BOM, and parsing a v1.6 yields a valid v1.6 BOM.
> 
> Parsing `issue408.xml` (v1.6) and generating a v1.5 BOM from that gives a `license` node with `acknowledgment` attribute, which fails v1.5 validation.

Added the validations d4d561cad1a6db52431522490a3ece603612ac7a , and also added tests from 1.6 to 1.4 for both XML and JSON

**Review by nscuro — APPROVED** [review]

**Pull request merged**